styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -263,6 +266,11 @@ private boolean shouldExecute(RunState runState) {
   }
 
   private void sendDequeue(InstanceState instanceState) {
+    if (!dequeueRateLimiter.tryAcquire()) {
+      LOG.debug("Dequeue rate limited");
+      return;
```

Comment:
Good catch. I put the rate limiting inside `sendDequeue` because of the code path fork in [tick()]( https://github.com/spotify/styx/pull/264/files/672e94bc80a01b2bdb1a31519b422e1b1be17f70#diff-a473be0c753046d2e7f56a00c3ed5b62R172). Maybe we can eliminate that fork and have all dequeuing go through `evaluateResourcesForDequeue` and have rate limiting happen before the resource usage counters are modified.